tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -24,22 +24,22 @@ using namespace shogun;
 using namespace Eigen;
 #endif // HAVE_EIGEN3
 
-//function to create sparse matrix
-template<typename T>
-SGSparseMatrix<T> GenerateSparseMatrix(const int32_t numberOfFeatures,
-                                       const int32_t numberOfVectors,
-                                       const float64_t sparseLevel)
+//function can generate both sparse and dense matrix accroding to sparse convention
+template<class MatrixType>
+void GenerateMatrixAccordingToSparseConvention(const float64_t &sparseLevel,
+                                               const int32_t   &numberOfFeatures,
+                                               const int32_t   &numberOfVectors,
+                                               const int32_t   &randSeed,
+                                               MatrixType* matrix)
 {
-    CRandom randGenerator(0);
-    SGSparseMatrix<T> matrix(numberOfFeatures, numberOfVectors);
+    CRandom randGenerator(randSeed);
     for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
         for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
         {
             float64_t randomNumber=randGenerator.random(0.0,1.0);
             if (randomNumber<=sparseLevel)
-                matrix(vecIndex,featIndex)=randomNumber;
+                (*matrix)(vecIndex,featIndex)=randomNumber*100;
         }
-    return matrix;
 }
```

Comment:
In case you didnt notice my previous conversation @iglesias 

The problem is if we initialize matrix as "MatrixType matrix(m,n)" and later use matrix(i,j) to access its contents. For sparse matrix, i goes from 0 to n-1, and j is from 0 to m-1, while dense matrix is i: 0 to m-1, j: 0 to n-1.

This is how sparse and dense matrix is currently implemented and what I mean the sparse and dense matrix use different interfaces.